### ISSUE TITLE
http://+ endpoints are not resolved properly in a cluster

### ISSUE THREAD

**Issue by cwe1ss** [c3-ls/ServiceFabric-Http#11]
Service Fabric hosts ASP.NET Core RC1 projects on endpoints like e.g. http://+:30001

It seems like this was done to allow us to call the service on multiple addresses in our dev cluster (localhost, 127.0.0.1, local IP).
However, since the `ServicePartitionResolver` always returns this address 1:1 when a fabric uri is resolved, this also results in "http://+:30001" in a multi-machine cluster. This means, the HttpClient always tries to access the service on the current machine.

I created [a question in the MSDN forums](https://social.msdn.microsoft.com/Forums/en-US/d127064a-82fd-45e6-803c-56778c72b56d/aspnet-5-projects-with-http-endpoints-cant-be-resolved-in-a-cluster?forum=AzureServiceFabric) because this looks like a bug to me.

**Comment by cwe1ss** [comment]
I'm pretty sure this is a bug. We now created our own version of Microsoft.ServiceFabric.AspNet.Hosting with a fix for this. We now always register the services with the actual IP address. This is now similar to the OWIN hosting scenario.

We fixed this in AspNetService.cs by changing the url assignment to this:
`_url = $"{endpoint.Protocol}://{FabricRuntime.GetNodeContext().IPAddressOrFQDN}:{endpoint.Port}";`

I'm not sure if we are allowed to put the whole thing on Github/Nuget since the original code isn't open source. I will try to ask someone.
